cmd/snap-seccomp/main_test.go

```diff
@@ -406,15 +389,33 @@ func (s *snapSeccompSuite) TestCompile(c *C) {
 		{"chown - u:root g:shadow", fmt.Sprintf("chown;native;-,0,%d", shadowGid), main.SeccompRetAllow},
 		{"chown - u:root g:shadow", fmt.Sprintf("chown;native;-,99,%d", shadowGid), main.SeccompRetKill},
 	} {
-		// skip socket tests if the system uses socketcall instead
-		// of socket
-		if strings.Contains(t.seccompWhitelist, "socket") && systemUsesSocketcall() {
-			continue
-		}
 		s.runBpf(c, t.seccompWhitelist, t.bpfInput, t.expected)
 	}
 }
 
+// TestCompileSocket runs in a separate tests so that only this part
+// can be skipped when "socketcall()" is used instead of "socket()".
+//
+// Some architectures (i386, s390x) use the "socketcall" syscall instead
+// of "socket". This is the case on Ubuntu 14.04, 17.04, 17.10
```

Comment:
17.04 and 17.10 shouldn't be using socketcall on i386 and s390x. The kernel and glibc should only be using socket and failures here would indicate a regression on those releases. Three pieces are needed, the kernel, glibc and libseccomp 2.3 for this to work. On 16.04, all this was in place-- why did this regress?

References:
* https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=809556 (i386 fixes)
* https://bugs.launchpad.net/ubuntu/+source/libseccomp/+bug/1554098 (s390x fixes)
*  https://git.kernel.org/cgit/linux/kernel/git/tip/tip.git/commit/?h=x86/asm&id=9dea5dc921b5f4045a18c63eb92e84dc274d17eb (kernel commit to 4.3)
* https://bugs.launchpad.net/ubuntu/+source/libseccomp/+bug/1576066